proversity_reports_script/report_backend/enrollment_report.py

```diff
@@ -0,0 +1,274 @@
+    def json_report_to_csv(self, json_report_data):
+        """
+        Process json data to convert into csv format.
+        """
+        pass
+
+    def create_csv_file(self, file_name, body_dict, headers, spreadsheet_id):
+        """
+        Creates the csv file with the passed arguments, and then save it locally.
+        """
+        pass
+
+    def upload_file_to_storage(self, course, path_file):
+        """
+        Uploads the csv report, to the implemented storage.
+        """
+        pass
+
+    def _get_salesforce_data(self, course, course_data):
+        """
+        Returns the Salesforce data according to the API specification.
+
+        A list of lists will be returned
+        and the size of the items in the list will be calculated from the BATCH_SIZE number.
+
+        Args:
+            course: course id string.
+            course_data: The enrollment report data returned from the platform.
+        Returns:
+            salesforce_data: List of lists with a size predifined by the BATCH_SIZE number.
+        """
+        def get_first_and_last_name(full_name):
+            """
+            Takes the argument full_name and returns the first name and last name separately.
+            If the full name value cannot be splitted at least once, it will return
+            the full_name argument as name and last name.
+
+            Args:
+                full_name: The student full name.
+            Returns:
+                name: The name of the student.
+                last_name: The last name of the student.
+            """
+            result = full_name.split(' ', 1)
+
+            if len(result) == 2:
+                return result
+
+            return full_name, full_name
+
+        salesforce_data = []
+        batch = []
+
+        for index, user in enumerate(course_data, 1):
+            # If the user has already a contact id,
+            # it's not neccesary to request the creation again.
+            if user.get('contact_id', ''):
+                continue
+
+            # If the user doesn't have a user_id, the creation cannot be possible.
+            if not user.get('user_id', ''):
+                continue
+
+            first_name, last_name = get_first_and_last_name(user.get('full_name', ''))
+            username = user.get('username', '')
+
+            user_salesforce_data = {
+                'FirstName': first_name if first_name else username,
+                'LastName': last_name if last_name else username,
+                'Email': user.get('email', ''),
+                'Company': self.extra_data.get('COMPANY_NAME', ''),
+                'Institution_Hidden': self.extra_data.get('INSTITUTION_HIDDEN_PREFIX', ''),
+                'Type_Hidden': self.extra_data.get('TYPE_HIDDEN', ''),
+                'Program_of_Interest': self.extra_data.get('PROGRAM_OF_INTEREST', ''),
+                'Intake_of_Intent': user.get('intake_of_intent', ''),
+                'Lead_Source': DEFAULT_LEAD_SOURCE,
```

Comment:
```suggestion
                'Lead_Source': self.extra_data.get('LEAD_SOURCE', DEFAULT_LEAD_SOURCE)
```
and add lead_source to the readme file

Reply:
Addresed.